src/field_theory/minpoly/gcd_monoid.lean

```diff
@@ -54,6 +54,7 @@ begin
   { exact (monic hs).map _ }
 end
 
+/- TODO: see if the `no_zero_smul_divisors S L` assumption can be lifted -/
```

Comment:
`L` is a field, right? So this should be automatic.

Reply:
Don't we need to assume `S` is a field too for this to be automatic?

Reply:
Ah yes, I switched `S` and `L`.

Reply:
It might be possible to fiddle around to get rid of the assumption but I'll have to think about it

Reply:
Take `R = Z`, `K = L = Q` and `S = Q[Y]` where the `S`-algebra structure on `L` is given by mapping `Y` to `0`. Now take `s = Y`: we have that `minpoly K (algebra_map S L s) = minpoly K 0 = X`, but `(minpoly R s).map (algebra_map R K) = 0` since `s` is not integral. Of course it would more interesting to find an example with `s` integral...

Reply:
My hope was that this would be removed using the fact that `s` is integral somehow, but I'm not sure this can be done :(

Reply:
If we take `R = ℤ`, `K = ℚ`, `S = ℤ[Y]/((Y^2-2)*(Y^2-3)) ` and `L = ℚ(√2)`, where the `S`-algebra structure on `L` is given by `Y ↦ √2`, with `s = Y`, it doesn't work.

A more natural assumption would be `is_domain S`, but I have to think about it.

Reply:
I am now a little bit lost and I don't know if this result is still needed, but I think the proof we had for `gcd_monoid` should now generalize, showing that `is_domain` is enough.